backend/de.metas.marketing/base/src/main/java/de/metas/marketing/base/process/MKTG_ContactPerson_ProcessBase.java

```diff
@@ -93,13 +94,18 @@ public void createContactPersonsForPartner(final MKTG_ContactPerson_ProcessParam
 				.addEqualsFilter(I_MKTG_Campaign_ContactPerson.COLUMN_MKTG_Campaign_ID, campaignId)
 				.create();
 
-		final Stream<User> usersToAdd = queryBL
+		final PInstanceId selectionId = queryBL
 				.createQueryBuilder(I_C_BPartner.class)
 				.addOnlyActiveRecordsFilter()
 				.filter(params.getSelectionFilter())
 				.andCollectChildren(I_AD_User.COLUMNNAME_C_BPartner_ID, I_AD_User.class)
 				.addNotInSubQueryFilter(I_AD_User.COLUMNNAME_AD_User_ID, I_MKTG_Campaign_ContactPerson.COLUMNNAME_AD_User_ID, linkTableQuery)
 				.create()
+				.createSelection();
+
+		final Stream<User> usersToAdd = queryBL.createQueryBuilder(I_AD_User.class)
+				.setOnlySelection(selectionId)
+				.create()
 				.setOption(IQuery.OPTION_GuaranteedIteratorRequired, false)
```

Comment:
actually you would get the same result with 
```suggestion
				.setOption(IQuery.OPTION_GuaranteedIteratorRequired, true)
```

to keep it simpler i think you should rather go with that